src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,52 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.Commands;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+
+/**
+ * A Parser to take in user input and determines the Command given to Duke.
+ */
+public class Parser {
+
+    /**
+     * Parse user input and returns an enum of the command.
+     *
+     * @param fullCommand is the user input from the terminal.
+     * @return an enum representing the command from the user.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        assert fullCommand instanceof String : "Command must be a String";
+        String[] splitString = fullCommand.split(" ");
+        Commands c;
```

Comment:
Quite confused with the difference between 'Command' and 'Commands' classes here, maybe another name for Commands like CommandTypes will be clearer?